game/src/main/scala/io/aigar/game/Virus.scala

```diff
@@ -0,0 +1,48 @@
+package io.aigar.game
+
+import com.github.jpbetz.subspace.Vector2
+import io.aigar.game.serializable.Position
+
+object Virus {
+  final val Quantity = 15
+  final val Mass = 100
+}
+
+class Virus(var position: Vector2) {
+
+  def update(grid: Grid, players: List[Player]): Unit = {
+    if(detectCollisions(players)) {
+      // TODO Add the virus consumption into the cell
+      respawn(grid, players)
+    }
+  }
+
+  def detectCollisions(players: List[Player]): Boolean ={
+    for(player <- players) {
+      for(cell <- player.cells) {
+        // TODO Change the 1.1 value to the constant
+        if(cell.contains(position) && cell.mass > Virus.Mass * 1.1){
```

Comment:
I don't know if you want to do this in this PR or if we should do it later (e.g. with #260), but it would be nice to refactor `Cell`/ `Virus` to have a common trait, like `Entity`, that would have `radius` and `mass`.

Then, we could add `isEating` to `Cell` and call it both here and in `Cell.eat`. The logic for the mass ratio and *"almost fully overlaps"* would be shared.